pkg/manager/impl/workflow_manager.go

```diff
@@ -161,16 +161,22 @@ func (w *WorkflowManager) CreateWorkflow(
 	}
 
 	// Assert that a matching workflow doesn't already exist before uploading the workflow closure.
-	existingMatchingWorkflow, err := util.GetWorkflowModel(ctx, w.db, *request.Id)
+	existingWorkflowModel, err := util.GetWorkflowModel(ctx, w.db, *request.Id)
 	// Check that no identical or conflicting workflows exist.
 	if err == nil {
 		// A workflow's structure is uniquely defined by its collection of nodes.
-		if bytes.Equal(workflowDigest, existingMatchingWorkflow.Digest) {
-			return nil, errors.NewFlyteAdminErrorf(
-				codes.AlreadyExists, "identical workflow already exists with id %v", request.Id)
+		if bytes.Equal(workflowDigest, existingWorkflowModel.Digest) {
+			return nil, errors.NewWorkflowExistsIdenticalStructureError(ctx, &request)
 		}
-		return nil, errors.NewFlyteAdminErrorf(codes.InvalidArgument,
-			"workflow with different structure already exists with id %v", request.Id)
+		// A workflow exists with different structure
+		existingCtx := getWorkflowContext(ctx, request.Id)
+		existingWorkflow, err := util.GetWorkflow(existingCtx, w.db, w.storageClient, *request.Id)
```

Comment:
instead of calling get again on the existing workflow, can we just have `NewWorkflowExistsDifferentStructureError` take in a core.Identifier, since it doesn't need the full admin protobuf def?